src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -748,6 +748,16 @@ private function resolveServices($value, $file, $isParameter = false)
 
                 return new Reference($id);
             }
+            if ('yaml_file' === $value->getTag()) {
+                $rootDir = \dirname($file);
+                $file = \is_file($argument) ? $argument : "$rootDir/$argument";
+
+                if (!\is_file($file)) {
+                    throw new InvalidArgumentException("Unable to locate file \"$argument\". Please provide a path relative to \"$rootDir\" or an absolute path.");
+                }
+
+                return $this->yamlParser->parseFile($file, Yaml::PARSE_CONSTANT);
```

Comment:
The fact that the loader itself is parsing the file means that this feature is available only in the YamlFileLoader, not in other formats. That's quite unexpected (all other features can be configured in all formats).

Reply:
@stof are you suggesting creating a new kind of `Symfony\Component\DependencyInjection\Argument\ArgumentInterface`?

Reply:
Well in PHP this can be easily done by parsing thanks to the YAML component directly, but `xml` would need a new argument type too I guess.

Reply:
Well, using XML, my comment does not make sense anymore, `yaml_file` is better 👍